docs/ADR-014-Total-conversion-functions-conventions.md

```diff
@@ -0,0 +1,70 @@
+# Status
+
+- [ ] Proposed 2024-05-21
+
+# Context
+
+## The Problem
+In `cardano-api` we have multiple functions performing conversions between one value of the type to the other, for example:
+
+```haskell
+toShelleyScriptHash :: ScriptHash -> ScriptHash
+fromShelleyScriptHash :: ScriptHash -> ScriptHash
```

Comment:
The argument type and return type are the same?

Reply:
I think it would be good to some more examples to demonstrate the breadth of applicability.

Reply:
Yep, include some examples so I can better understand what we're looking at.